src/AI/Attack.pm

```diff
@@ -153,7 +153,25 @@ sub process {
 		main();
 	}
 
-	# Check for kill steal and mob-training while moving
+	# Check for hidden monsters
+	if (AI::inQueue("attack") && AI::is("move", "route", "attack")) {
+		my $ID = AI::args->{attackID};
+		my $monster = $monsters{$ID};
+		if (($monster->{statuses}->{EFFECTSTATE_BURROW} || $monster->{statuses}->{EFFECTSTATE_HIDING}) && 
+		$config{avoidHiddenMonsters}) {
+			message TF("Dropping target %s - will not attack hidden monsters\n", $monster), 'ai_attack';
+			$char->sendAttackStop;
+			$monster->{ignore} = 1;
+
+			AI::dequeue while (AI::inQueue("attack"));
+			if ($config{teleportAuto_dropTargetHidden}) {
+				message T("Teleport due to dropping hidden target\n");
+				useTeleport(1);
```

Comment:
looks like we should fix it after #3717

Reply:
fixed at https://github.com/OpenKore/openkore/pull/3756/commits/4e25b0519fa97566f8313b1cc501c7b51dcd7cac